sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/DocumentResult.java

```diff
@@ -64,4 +66,15 @@ public TextAnalyticsError getError() {
     public boolean isError() {
         return isError;
     }
+
+    /**
+     * Throw a TextAnalyticsException if result has isError true and when a non-error property was accessed.
+     */
+    void throwExceptionIfError() {
+        if (this.isError()) {
+            throw logger.logExceptionAsError(new TextAnalyticsException(
+                String.format("Error in accessing the property when %s returned with an error.",
+                    this.getClass()), null, null));
```

Comment:
Would it make sense to return the document id/error message in the exception message here?